### PR TITLE
(dsl): Support 'hasChild' query

### PR DESCRIPTION
**Pull request by markaya**
Resolves `hasChildQuery` in #91 

Summary:
- Add support for creating `has_child` query
- Add tests for creation and encodin `has_child` query
- fix scaladoc for `HasParent` creation methods

### CODE REVIEW

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 1]
```diff
@@ -110,7 +109,36 @@ object ElasticQuery {
     Bool[Any](filter = queries.toList, must = Nil, mustNot = Nil, should = Nil, boost = None, minimumShouldMatch = None)
 
   /**
-   * Constructs an instance of [[zio.elasticsearch.query.HasParent]] using the specified parameters.
+   * Constructs an instance of [[zio.elasticsearch.query.HasChildQuery]] using the specified parameters.
+   *
+   * @param childType
+   *   a name of the child relationship mapped for the join field
+   * @param query
+   *   query you wish to run on child documents of the child `type` field
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:117]: A [[ElasticQuery]] ...

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 2]
```diff
@@ -110,7 +109,36 @@ object ElasticQuery {
...
+   * @tparam S
+   *   document for which field query is executed
+   * @return
+   *   an instance of [[zio.elasticsearch.query.HasChildQuery]] that represents the has parent query to be performed.
+   */
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:122]: ..has child query..

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 3]
```diff
@@ -110,7 +109,36 @@ object ElasticQuery {
...
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.HasChildQuery]] using the specified parameters.
+   *
+   * @param childType
+   *   a name of the child relationship mapped for the join field
+   * @param query
+   *   query you wish to run on child documents of the child `type` field
+   * @return
+   *   an instance of [[zio.elasticsearch.query.HasChildQuery]] that represents the has parent query to be performed.
+   */
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:136]: ..has child query..

File: modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala [hunk 4]
```diff
@@ -110,7 +109,36 @@ object ElasticQuery {
...
+  final def hasChild[S: Schema](childType: String, query: ElasticQuery[S]): HasChildQuery[S] =
+    HasChild(childType = childType, query = query)
+
+  /**
+   * Constructs an instance of [[zio.elasticsearch.query.HasChildQuery]] using the specified parameters.
+   *
+   * @param childType
+   *   a name of the child relationship mapped for the join field
+   * @param query
+   *   query you wish to run on child documents of the child `type` field
```
Comment (dbulaja98) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticQuery.scala:133]: A [[ElasticQuery]] ...

File: modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala [hunk 5]
```diff
@@ -109,6 +109,75 @@ private[elasticsearch] final case class Exists[S](field: String) extends ExistsQ
...
+   *
+   * @param value
+   *   the [[scala.Int]] value for `score` parameter
+   * @return
+   *   a new instance of the [[HasChildQuery]] with the `score` value set.
+   */
+  def maxChildren(value: Int): HasChildQuery[S]
+
+  /**
+   * Sets the `minChildren` parameter parameter for the [[HasChildQuery]].
```
Comment (drmarjanovic) [inline modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala:132]:
```suggestion
   * Sets the `minChildren` parameter for the [[HasChildQuery]].
```

**Review by drmarjanovic — CHANGES_REQUESTED** [review]

**Review by drmarjanovic — APPROVED** [review]

**Pull request merged**